### ISSUE TITLE
Add online tensor LDA (TensorLDA) — experimental tier

### ISSUE THREAD

**Issue by nealcaren** [nealcaren/topica#255]
## Summary

Port **online tensor LDA** (method-of-moments LDA via 2nd/3rd-order cumulants) as a pure-Rust, CPU, numpy-native model in the **experimental tier** (`topica.enable_experimental()`), validated against the reference Python package.

- Reference: [tensorly/tlda](https://github.com/tensorly/tlda) (Modified BSD)
- Paper: Kangaslahti et al., *Analyzing Political Text at Scale with Online Tensor LDA*, **Political Analysis 2026** ([arXiv:2511.07809](https://arxiv.org/abs/2511.07809))

**Status: parked until the release/API freeze lifts** (see `future` label). This is a plan, not a start. Strong audience fit (political text at scale; same journal as ECTM), and a fresh reference impl + paper to validate against.

## Why it's tractable

The "tensor decomposition" never materializes a tensor. The CP step is compact matmuls with elementwise squares — `cumulant_gradient` is two lines:

```
grad = 3(1+θ)·Φ·(ΦᵀΦ)²  −  3(1+α₀)(2+α₀)/(2N)·Yᵀ·(YΦ)²     # ² elementwise
Φ -= lr·grad ; normalize columns
```

The algorithm in four pieces:
1. **Whiten (2nd-order cumulant):** scale counts by `√(α₀+1)`, center, top-K eigenpairs `(U,λ)`; `W = U/√λ`, `transform(X)=X·W`, `reverse(Z)=Z·(U·√λ)ᵀ`.
2. **CP factors via SGD (3rd-order):** `Φ` (K×K), random→QR init, the gradient loop above to a max-abs-change tol.
3. **Recover:** column norms → eigenvalues → `α` + topic weights; unwhiten `Φ` → topic-word `β`; simplex-project.
4. **Infer doc-topic (`predict`):** per-doc Dirichlet posterior (gamma-init fixed point).

The only primitive topica lacks is a **truncated symmetric eigensolver / thin SVD** for whitening (we have Cholesky/`spd_inverse`, no eig/SVD). Everything else is `ndarray` matmuls already in the codebase.

## PR breakdown (one at a time off `main`; each green on fmt + clippy -D + tests)

1. **linalg primitives** — `randomized_svd`/`truncated_eig` + reduced `qr` in `topica-core/src/linalg.rs`, golden-tested vs numpy. (Worth a quick spike first to confirm randomized-SVD accuracy is enough for whitening.)
2. **core estimator (batch)** — `tlda.rs`: `fit(counts, k, alpha_0, …)` = whiten → SGD CP → recover β/α. Rust test recovers planted topics (mirrors `spectral.rs`); gradient FD-checked vs `loss_rec`.
3. **Python binding + experimental gate** — `TensorLDA` pyclass (own module, `use super::*`), gated by the `enable_experimental()` AtomicBool like ECTM; `fit`/`transform`/`top_words`/`topic_word`/`doc_topic`, `.pyi` stub, conformance entry.
4. **online `partial_fit`** — incremental whitening (running mean + incremental PCA) + per-batch SGD, matching tlda's `partial_fit(X_batch, batch_index)`. This is the paper's headline (scale without holding the matrix).
5. **parity + docs** — `parity/tlda_compare.py` driving the `tlda` package (optional dep, skip-clean) on a shared corpus (their `plot_uci_topics`, or poliblog); aligned topic-word cosine vs the reference and vs our Gibbs LDA. Experimental-tier docs page.

## Validation bar
- Recovers planted topics on synthetic LDA (Rust test).
- Aligned topic-word cosine vs the `tlda` package within seed noise on a shared corpus; sanity vs topica's Gibbs LDA.
- Deterministic for a fixed seed.

## Risks / scope
- **Whitening conditioning:** `√λ` divide blows up on tiny eigenvalues → eigenvalue floor, documented.
- **SGD hyperparameters** (`learning_rate`, `n_iter_train`, `θ`): port their defaults, expose them, no auto-tuning claim.
- **Not GPU / not 1B-docs:** our value-add is CPU, dependency-light, deterministic MoM-LDA; the streaming path (PR4) is what makes it useful at scale on CPU. Set expectations in docs.
- **No PyTorch/tensorly dependency** — pure Rust core, against topica's design otherwise.

## Effort
~the size of adding a new spectral family. PR1–PR2 are the real work (eigensolver + SGD loop); PR3–PR5 are mechanical given the existing add-model recipes.
